gremlin-go/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ARG GREMLIN_SERVER_VERSION=${version:-3.5.4}
+FROM tinkerpop/gremlin-server:${GREMLIN_SERVER_VERSION}-SNAPSHOT
```

Comment:
i'm not sure i understand how this works as docker is still something i'm learning. i understand what an `ARG` is but i don't follow what the implications are of hardcoding to 3.5.4 and then appending "-SNAPSHOT". what happens when we are doing build artifacts on 3.5.4 and we are no longer on "-SNAPSHOT"? Isn't there a nice way to gather the version from the `pom.xml` and pass it through so that all of this stays dynamic?